Update supported modules table

stmoy opened the pull request:
This change updates the docs to include the recently supported modules, and expands the previous table to also discuss macOS support. It also adds a note about reactnative.directory.

Fixes #202 

# Open Questions

- Is it worthwhile to have the specific "Supported Version" columns or could this be collapsed to ✔ vs ❌ ?
- Is the macOS version column correct?
- Are we missing any modules?
- This doc currently lives in the Native Modules (Windows) section of the sidebar; is that correct or should it move?

tom-un commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+| <ins>[datetimepicker](https://github.com/react-native-community/datetimepicker)</ins> | 0.62 | ? |
+| <ins>[react-native-checkbox](https://github.com/react-native-community/react-native-checkbox)</ins> | 0.62 | ? |
+| <ins>[react-native-netinfo](https://www.github.com/react-native-community/react-native-netinfo)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-picker](https://github.com/react-native-community/react-native-picker)</ins> | 0.61 | 0.61 |
```
There is no picker in the community module.  An impl exists in react-native-macos but someone needs to port it to the community repo (https://github.com/microsoft/react-native-macos/issues/395)

tom-un commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+| <ins>[react-native-checkbox](https://github.com/react-native-community/react-native-checkbox)</ins> | 0.62 | ? |
+| <ins>[react-native-netinfo](https://www.github.com/react-native-community/react-native-netinfo)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-picker](https://github.com/react-native-community/react-native-picker)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-progress-view](https://github.com/react-native-community/progress-view)</ins> | 0.62 | ? |
```
There is a progress view in react-native-macos but someone needs to port it to the community module (https://github.com/microsoft/react-native-macos/issues/391)

tom-un commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+| <ins>[react-native-netinfo](https://www.github.com/react-native-community/react-native-netinfo)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-picker](https://github.com/react-native-community/react-native-picker)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-progress-view](https://github.com/react-native-community/progress-view)</ins> | 0.62 | ? |
+| <ins>[react-native-slider](https://github.com/react-native-community/react-native-slider)</ins> | 0.62 | ? |
```
There is a slider in react-native-macos but someone needs to port it to the community module (https://github.com/microsoft/react-native-macos/issues/394)

tom-un commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+| Name | Version Supported (Windows) | Version Supported (macOS) |
+|:-|:-|:-|
+| <ins>[react-native-camera](https://www.github.com/react-native-community/react-native-camera)</ins> | 0.61 | x |
+| <ins>[react-native-device-info](https://www.github.com/react-native-community/react-native-device-info)</ins> | 0.61 | 0.61 |
```
There is a *draft* PR open to add Device Info support for macOS https://github.com/react-native-community/react-native-device-info/pull/1057

tom-un commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+|:-|:-|:-|
+| <ins>[react-native-camera](https://www.github.com/react-native-community/react-native-camera)</ins> | 0.61 | x |
+| <ins>[react-native-device-info](https://www.github.com/react-native-community/react-native-device-info)</ins> | 0.61 | 0.61 |
+| <ins>[react-native-navigation](https://github.com/react-navigation/react-navigation)</ins> | 0.62 | x |
```
react-navigation has at least partial macOS support via https://github.com/react-navigation/react-navigation/pull/8570

acoates-ms commented on website/versioned_docs/version-0.62/supported-community-modules.md:
```diff
@@ -0,0 +1,50 @@
+- [Modules that support Windows](https://reactnative.directory/?windows=true)
+- [Modules that support macOS](https://reactnative.directory/?macos=true)
+
+## Microsoft-supported Modules
```
I think the word supported here is the one I'd be careful with.

Maybe "Modules with Microsoft contributed Windows and MacOS implementions"?

Maybe liberally replace the use of support with implementation throughout the doc?

stmoy replied:
Did the thing

stmoy commented:
I've addressed all of the feedback so I'm going to merge this change in prior to the expected release next week. If there is more feedback we can always update the docs later 😉
